lucene/core/src/java/org/apache/lucene/search/TopFieldCollector.java

```diff
@@ -414,6 +428,45 @@ public static TopFieldCollector create(Sort sort, int numHits, FieldDoc after,
     }
   }
 
+  /**
+   * Same as above with an additional parameter to allow passing in the threshold checker
+   */
+  public static TopFieldCollector create(Sort sort, int numHits, FieldDoc after,
+                                         int totalHitsThreshold, HitsThresholdChecker hitsThresholdChecker) {
```

Comment:
We don't need the `totalHitsThreshold` here, is it possible to move the argument validation (totalHitsThreshold < 0) in the HitsThresholdChecker ctr ?